### PR TITLE
Validate geo json

### LINKED ISSUE #3: Validación de polígono en endpoints
# Descripción

Así como se creó documentación y validación para los parámetros `metric_id`, `area_type` y `area_id` en los endpoints del nuevo backend, crear documentación, validación y ejemplos para validar un polígono en GeoJSON.

Adicionalmente agregar los ejemplos de salida para los códigos 200 y 422 para todos los endpoints

# Consideraciones

Revisar la documentación de FastAPI y las librerías en las que se apoya (Pydantic y typing), tanto para la validación como para la documentación de los ejemplos de las respuestas.

Así mismo, si es necesario, se puede mover la clase [Polygon](https://github.com/PEM-Humboldt/biotablero-search-backend/blob/setup/app/routers/metrics.py#L33) a otro archivo, esta se creó con este fin

Sacar la rama de esta tarea de la rama setup (si aún no se ha mezclado en develop)

### PR DESCRIPTION
**Pull request by ManuelStardust**
## 🛠️ Changes
- Add validation for GeoJson input
- Add 200 and 422 examples

## 📝 Associated issues
#3 

## 🤔 Considerations
You can test the API response with this examples for Insomnia [Backend Python.json](https://drive.google.com/file/d/1ChkZVvoHP0pzEI_rgDv7CvMBADn1okhA/view?usp=drive_link)

### CODE REVIEW

File: app/routers/polygon.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    [-66.851117, 4.214931],
+    [-66.851117, 12.437303],
+    [-78.990935, 12.437303],
+]
+
+geojson_polygon = {
+    "type": "Feature",
+    "properties": {},
+    "geometry": {"type": "Polygon", "coordinates": [coordinates]},
+}
```
Comment (cazapatamar) [inline app/routers/polygon.py]: Should we separate this example polygon in other file?

**Review by cazapatamar — COMMENTED** [review]
Im getting ´Internal server Error´ in the endpoint ´/metrics/{metric_id}/areas´ (get and post)

imagen

imagen

File: app/routers/metrics.py [hunk 2]
```diff
@@ -54,11 +70,11 @@ async def get_areas_by_defined_area(
     ]
 
 
-@router.post("/{metric_id}/areas")
+@router.post("/{metric_id}/areas", response_model=List[AreasResponse])
 async def get_areas_by_polygon(
     metric_id: Annotated[str, fastapi.Depends(metric_id_param)],
     polygon: Polygon,
-):  # TODO: Define return type
+) -> list[dict[str, float]]:  # TODO: Define return type
```
Comment (erikasv) [inline app/routers/metrics.py]: remove todo

File: app/routers/polygon.py (hunk 1 above)
Comment (erikasv) [reply app/routers/polygon.py:16]: What would be the benefit?

File: app/routers/polygon.py
Comment (erikasv) [inline app/routers/polygon.py:1]: I don´t like this class here, `app/routers/polygon` seem to be routes for polygons, but this class is for the Polygon object itself. I don't remember if we agreed in some pattern, if not, propose something to improve organization.

Comment (erikasv) [reply app/routers/polygon.py:1]: I've seen this objects called schemas or DTO (data transfer object). I think schema is ok, and they could be in a folder inside routers

Comment (erikasv) [reply app/routers/polygon.py:1]: Whatever you decide, document it in the "Proyectos en python" file

File: app/routers/polygon.py (hunk 1 above)
Comment (ManuelStardust) [reply app/routers/polygon.py]: We decided to keep this example in the polygon file because it is a short example.

**Review by erikasv — APPROVED** [review]

**Review by cazapatamar — APPROVED** [review]

**Pull request merged**